fdb-record-layer-lucene/src/main/java/com/apple/foundationdb/record/lucene/LuceneIndexMaintainer.java

```diff
@@ -232,10 +279,22 @@ private void deleteDocument(Tuple groupingKey, byte[] primaryKey) throws IOExcep
             throw new RecordCoreException("Issue deleting old index keys", "oldRecord", oldRecord, e);
         }
 
+        Map<Tuple, AnalyzingInfixSuggester> suggesterMap = new HashMap<>();
+
         // update new
         try {
             for (Map.Entry<Tuple, List<LuceneDocumentFromRecord.DocumentField>> entry : newRecordFields.entrySet()) {
-                writeDocument(entry.getValue(), entry.getKey(), newRecord.getPrimaryKey().pack());
+                if (autoCompleteEnabled) {
+                    final AnalyzingInfixSuggester suggester = getSuggester(entry.getKey());
+                    suggesterMap.put(entry.getKey(), suggester);
+                    writeDocument(entry.getValue(), entry.getKey(), newRecord.getPrimaryKey().pack(), suggester);
```

Comment:
Since the primary key is getting saved possible elsewhere split up we probably don't want to add the suggester to it. It seems like that could expose problems if the user didn't intend it to be searched along with the indexed fields.

Reply:
If the fields in the primary key are included in the indexed fields they'll be added to the suggester individually right?